releasenotes/notes/fix-register-name-format-deprecation-61ad5b06d618bb29.yaml

```
@@ -0,0 +1,8 @@
+---
+fixes:
+  - |
+    Fixed a bad deprecation of :attr:`.Register.name_format` which had made the class attribute
+    available only from instances and not the class.  When trying to send dynamic-circuits jobs to
+    hardware backends, this would frequently cause the error:
+
+      AttributeError: 'property' object has no attribute 'match'
```

Do you want to add a bug reference here like:

```suggestion
      AttributeError: 'property' object has no attribute 'match'

    Fixed `#9493 <https://github.com/Qiskit/qiskit-terra/issues/9493>`__
```